src/Microsoft.ML.Data/DataView/InternalSchemaDefinition.cs

```diff
@@ -141,17 +141,19 @@ private InternalSchemaDefinition(Column[] columns)
         /// <param name="itemType">
         /// The corresponding <see cref="PrimitiveDataViewType"/> RawType of the type, or items of this type if vector.
         /// </param>
-        public static void GetVectorAndItemType(MemberInfo memberInfo, out bool isVector, out Type itemType)
+        /// <param name="data">DataView that contains the expected type of input and output of transformers</param>
+        /// <param name="singleName">Column name used to find the corresponding expected data type</param>
+        public static void GetVectorAndItemType(MemberInfo memberInfo, out bool isVector, out Type itemType, IDataView data = null, string singleName = null)
         {
```

Comment:
Wasn't this the original problem we were trying to solve? To not include IDataView dependency within SchemaDefinition?  Can we implement this logic back in TypedCursor or refactor it differently?

Reply:
It seems I didn't realize this function is in SchemaDefinition. Will change that..